src/bin/cli.rs

```diff
@@ -1,25 +1,30 @@
 use mini_redis::{client, DEFAULT_PORT};
 
 use bytes::Bytes;
+use clap::{Parser, Subcommand};
 use std::num::ParseIntError;
 use std::str;
 use std::time::Duration;
-use structopt::StructOpt;
 
-#[derive(StructOpt, Debug)]
-#[structopt(name = "mini-redis-cli", version = env!("CARGO_PKG_VERSION"), author = env!("CARGO_PKG_AUTHORS"), about = "Issue Redis commands")]
+#[derive(Parser, Debug)]
+#[clap(
+    name = "mini-redis-cli",
+    version,
+    author,
+    about = "Issue Redis commands"
+)]
 struct Cli {
-    #[structopt(subcommand)]
+    #[clap(subcommand)]
     command: Command,
 
-    #[structopt(name = "hostname", long = "--host", default_value = "127.0.0.1")]
+    #[clap(name = "hostname", long = "--host", default_value = "127.0.0.1")]
     host: String,
 
-    #[structopt(name = "port", long = "--port", default_value = DEFAULT_PORT)]
+    #[clap(name = "port", long = "--port", default_value = DEFAULT_PORT)]
```

Comment:
nit: i don't _think_ the `name` key is necessary here, since the name of the field is the same as the `name` value:
```suggestion
    #[clap(long = "--port", default_value = DEFAULT_PORT)]
```